spec/query_spec.cr

```diff
@@ -91,6 +91,25 @@ describe LuckyRecord::Query do
       count.should eq 1
     end
   end
+
+  describe "#not" do
+    it "negates the given where condition as 'equal'" do
+      insert_a_user
+
+      results = UserQuery.new.name.not("not the name").results
+      results.should eq UserQuery.new.results
+
+      results = UserQuery.new.name.not("Paul").results
+      results.should eq [] of User
```

Comment:
I'd love to see one extra spec with extra changing, and maybe some updates to the `insert_a_user` method to make it more clear why these specs pass. What do you think about this?

```crystal
    it "negates the given where condition" do
      insert_a_user(name: "Paul")

      results = UserQuery.new.name.not("not the name").results
      results.map(&.name).should eq ["Paul"]

      results = UserQuery.new.name.not("Paul").results
      results.should eq [] of User

      insert_a_user(name: "Jill")
      insert_a_user(name: "Steve")
      results = UserQuery.new.name.lower.not("steve").results
      results.map(&.name).should eq ["Paul", "Jill"]
    end

private def insert_a_user(age = 34, name = "Paul")
  LuckyRecord::Repo.run do |db|
    db.exec "INSERT INTO users(name, created_at, updated_at, age, joined_at) VALUES ($1, $2, $3, $4, $5)",
      name,
      Time.now,
      Time.now,
      age,
      Time.now
  end
end
```